src/ScillaParser.ts

```diff
@@ -173,3 +178,31 @@ const extractTransitions = (ccompsElem: any[]): Transitions => {
     };
   });
 };
+
+function parseField(row : any):Field{
+  const field_type = row[0];
+
+  if (field_type == "PrimType"){
+    const type = row[1];
+    return {
+      name: "",
+      typeobject: type,
+      type: type
+    }
+  } else if (field_type == "ADT"){
+    const ctor = row[1][1][1];
+    const argtypes = row[2].map(parseField);
+    const name = row[0][1][1];
+    const ADT : ADTField = {
+      ctor: ctor,
+      argtypes: argtypes
+    }
+    return {
+      name: name,
+      typeobject: ADT,
+      type: ctor + argtypes.map((arg:Field)=>` (${arg.type})`).join('')
```

Comment:
This is slightly different because of the preceding space. Admittedly it looks weird because of having to specify no space in the join. Would it be better to have `ctor + " " + argtypes.map((arg: Field) => arg.type).join(' ')`?

Reply:
IMO that would be clearer yeah :)